src/actions/attack.cpp

```diff
@@ -1552,15 +1552,13 @@ void attack_unit_and_advance(const map_location& attacker,
 	}
 }
 
-std::pair<int, map_location> under_leadership(const unit_map& units, const map_location& loc, const_attack_ptr weapon, const_attack_ptr opp_weapon)
+int under_leadership(const unit &u, const map_location& loc, const_attack_ptr weapon, const_attack_ptr opp_weapon)
 {
-	const unit_map::const_iterator un = units.find(loc);
-	if(un == units.end()) {
-		return {0, map_location::null_location()};
-	}
+	unit_ptr new_unit = u.clone();
```

Comment:
Would it suffice to only clone the unit if `u.get_location() != loc`? Or should clone be avoided entirely (by passing `loc` to `get_abilities`, I suppose)?

Reply:
i think passing log to get_abilities is the better slution, inparticular since it seems like it does already support that https://github.com/wesnoth/wesnoth/blob/6ef7dd77ab7f3e64c74f1153a00b332361197f37/src/units/unit.hpp#L1492